js/src/data/test/reducer.test.js

```diff
@@ -349,6 +349,27 @@ describe( 'reducer', () => {
 			expect( state ).toHaveProperty( path, action.adsCampaigns );
 		} );
 
+		it( 'should push the created ads campaign and return with updated ads campaigns', () => {
+			const createdToInitialState = reducer( prepareState(), {
+				type: TYPES.CREATE_ADS_CAMPAIGN,
+				createdCampaign: { id: 123 },
+			} );
+			const createdToLoadedState = reducer( createdToInitialState, {
+				type: TYPES.CREATE_ADS_CAMPAIGN,
+				createdCampaign: { id: 456 },
+			} );
+
+			createdToInitialState.assertConsistentRef();
+			createdToLoadedState.assertConsistentRef();
+			expect( createdToInitialState ).toHaveProperty( path, [
+				{ id: 123 },
+			] );
+			expect( createdToLoadedState ).toHaveProperty( path, [
+				{ id: 123 },
+				{ id: 456 },
+			] );
+		} );
```

Comment:
Currently, the errors won't be passed to reducer. Do you mean the test cases for `createAdsCampaign` and `updateAdsCampaign` actions?

Assuming you were meaning actions, I'm not sure about that. Because currently the APIs do not respond with HTTP status 400 or something like that when the parameters are invalid (see #1210). For example, should `apiFetch` mock simulate the problem of current APIs, or should it simulate a non-existent HTTP status 400 response?

Given that neither of these is really certain, I thought we could wait for the #1210 API to be fixed and then test it accordingly to make it robust and less confusing.

Reply:
> Currently, the errors won't be passed to reducer. Do you mean the test cases for createAdsCampaign and updateAdsCampaign actions?

What I was looking for was to have a case in which the API call fails and we check at least that it is actually throwing an error.  As far as I understand this was one of the intends of this PR. 

> For example, should apiFetch mock simulate the problem of current APIs

I think the best would be to test always with the current status and then update the code and tests after the final fix.

> Given that neither of these is really certain, I thought we could wait for the #1210 API to be fixed and then test it accordingly to make it robust and less confusing.

I'm unsure when it is expected to have the issue fixed, but if you consider it is appropriate to wait, let's wait until the final fix is done.